CIDEVSTC-278 Adds inspection for data process definitions

lukecampbell opened the pull request:
- Adds method in data_process_management for inspecting the source code of a data process definition
- Adds test for testing the inspection of a well-known dpd and source code
- Refactors TestDMExtended's SA-based tests into a new test class TestDataProcessFunctions

Depends on https://github.com/ooici/ion-definitions/pull/486

lukecampbell committed:
Adds implementation of inspect_data_process_definition

lukecampbell committed:
Adds test for inspect_data_process_definition

lukecampbell committed:
Small patch

lukecampbell committed:
Updates service definitions for dpms

- inspect_data_process_definition

lukecampbell commented:
Ready to merge.
here's the test on coi_any
http://buildbot.oceanobservatories.org:8010/builders/coi_any/builds/168

MauriceManning commented on ion/services/sa/process/data_process_management_service.py:
```diff
@@ -392,6 +394,45 @@ def create_data_process_new(self, data_process_definition_id='', in_data_product
+        else:
+            raise BadRequest("Data process type not yet supported")
+
+    def _inspect_parameter_function(self, data_process_definition_id):
```
this could just be inspect_function which checks the type of the DPD then follows the correct assoc to get either PF or TF. The logic should be the same for both.

lukecampbell replied:
There's a bit of a difference. The TF has all of the module and egg information defined in the resource. The PF has all the information defined in a serialized dictionary for the Coverage Model.

MauriceManning replied:
yes, agreed.

lukecampbell committed:
Creates associations for data process book keeping

- Adds an association for every data process for every parameter
  function.
- Fixes test_verify_service issue with inspect_data_process_definition

lukecampbell commented on ion/services/sa/product/data_product_management_service.py:
```diff
@@ -149,7 +149,9 @@ def create_data_processes(self, data_product_id=''):
-            self.clients.resource_registry.create(dp)
+            dp_id, _ = self.clients.resource_registry.create(dp)
+            self.clients.resource_registry.create_association(dpd._id, PRED.hasDataProcess, dp_id)
+            self.clients.resource_registry.create_association(dp_id, PRED.hasOutputProduct, data_product._id)
```
This is important.

For every parameter function in every data product:
- A data process is created. 
- The DPD is associated with the DP
- The DP is associated with the Data Product.
